zerver/actions/message_delete.py

```diff
@@ -45,7 +45,9 @@ def check_update_first_message_id(
     send_event_on_commit(realm, stream_event, users_to_notify)
 
 
-def do_delete_messages(realm: Realm, messages: Iterable[Message]) -> None:
+def do_delete_messages(
+    realm: Realm, messages: Iterable[Message], acting_user: UserProfile | None = None
```

Comment:
Let's use `*, ` to make require the `acting_user` parameter always be passed as a kwarg; that's a valuable pattern for readability in all our `do_` functions that accept such a parameter.

Reply:
Reworked this, which included noticing and fixing a bug that you'd not updated an important caller:

``` diff
diff --git a/zerver/actions/message_delete.py b/zerver/actions/message_delete.py
index 6983f1e790..70e909cc31 100644
--- a/zerver/actions/message_delete.py
+++ b/zerver/actions/message_delete.py
@@ -46,7 +46,7 @@ def check_update_first_message_id(
 
 
 def do_delete_messages(
-    realm: Realm, messages: Iterable[Message], acting_user: UserProfile | None = None
+    realm: Realm, messages: Iterable[Message], *, acting_user: UserProfile | None = None
 ) -> None:
     # messages in delete_message event belong to the same topic
     # or is a single direct message, as any other behaviour is not possible with
diff --git a/zerver/views/message_edit.py b/zerver/views/message_edit.py
index 5a974ef3b6..2e70b3a369 100644
--- a/zerver/views/message_edit.py
+++ b/zerver/views/message_edit.py
@@ -183,7 +183,7 @@ def delete_message_backend(
     message = access_message(user_profile, message_id, lock_message=True)
     validate_can_delete_message(user_profile, message)
     try:
-        do_delete_messages(user_profile.realm, [message], user_profile)
+        do_delete_messages(user_profile.realm, [message], acting_user=user_profile)
     except (Message.DoesNotExist, IntegrityError):
         raise JsonableError(_("Message already deleted"))
     return json_success(request)
diff --git a/zerver/views/streams.py b/zerver/views/streams.py
index 2deb5c5233..609bbd6f06 100644
--- a/zerver/views/streams.py
+++ b/zerver/views/streams.py
@@ -938,7 +938,7 @@ def delete_in_topic(
             )
             if not messages_to_delete:
                 break
-            do_delete_messages(user_profile.realm, messages_to_delete)
+            do_delete_messages(user_profile.realm, messages_to_delete, acting_user=user_profile)
 
     return json_success(request, data={"complete": True})
 
```